airflow/models/taskinstance.py

```diff
@@ -878,6 +878,19 @@ def refresh_from_db(self, session: Session = NEW_SESSION, lock_for_update: bool
         else:
             self.state = None
 
+    @classmethod
+    def get_from_key(cls, ti_key, session) -> TaskInstance | None:
```

Comment:
IMHO `get_by_key` is more suitable
```suggestion
    def get_by_key(cls, ti_key, session) -> TaskInstance | None:
```